### PR TITLE
Fix tedge-agent stuck on too many pending operations

### PR DESCRIPTION
**Pull request by didier-wenzek**
## Proposed changes

Introduce unbounded receivers. Given a cycle of actors (say A sending messages to B, B to C and C to A), at least one of these actors must use such an unbounded receiver.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://github.com/thin-edge/thin-edge.io/issues/2639

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/tedge_agent/src/tedge_operation_converter/actor.rs [hunk 1]
```diff
@@ -43,7 +43,7 @@ pub struct TedgeOperationConverterActor {
     pub(crate) workflows: WorkflowSupervisor,
     pub(crate) state_repository: AgentStateRepository<CommandBoard>,
     pub(crate) log_dir: Utf8PathBuf,
-    pub(crate) input_receiver: LoggingReceiver<AgentInput>,
+    pub(crate) input_receiver: UnboundedLoggingReceiver<AgentInput>,
```
Comment (didier-wenzek) [inline crates/core/tedge_agent/src/tedge_operation_converter/actor.rs:46]: An alternative to fix the issue could be to use two receivers.

- One receiver to receive progress feedback from the software actor
- Another to collect requests from MQTT

Using a biased select, one can then give the priority to feedback messages. 

 with a biased `select!`

**Review by rina23q — APPROVED** [review]
LGTM

File: tests/RobotFramework/tests/tedge_agent/main_tedge_agent.robot [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+Test Tags           theme:tedge_agent
+Test Setup          Custom Setup
+Test Teardown       Get Logs
+
+*** Test Cases ***
+
+On start process all the pending operations
+    [Tags]    \#2369
+    Execute Command    systemctl stop tedge-agent
```
[the remark this answers is not in the document]
Comment (didier-wenzek) [reply tests/RobotFramework/tests/tedge_agent/main_tedge_agent.robot]: Done

File: tests/RobotFramework/tests/tedge_agent/main_tedge_agent.robot [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    FOR  ${id}   IN RANGE    0   10
+         Execute Command    tedge mqtt pub --retain te/device/main///cmd/software_list/test-${id} '{"status":"init"}'
+    END
+    Execute Command    systemctl start tedge-agent
```
[the remark this answers is not in the document]
Comment (didier-wenzek) [reply tests/RobotFramework/tests/tedge_agent/main_tedge_agent.robot]: Done

**Pull request merged**